### PR TITLE
Upstream FIRRTL->FIRRTL2 Bridge

### PR DESCRIPTION
**Pull request by jerryz123**
This is taken from @ekiwi 's work on supporting chiseltest for chisel5+.

### Contributor Checklist

- [ ] Did you add at least one test demonstrating the PR?
- [ ] Did you delete any extraneous printlns/debugging code?
- [ ] Did you specify the type of improvement?

#### Type of Improvement

 - new feature/API

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -149,6 +149,16 @@ lazy val firrtl = (project in file("."))
...
   .settings(mimaSettings)
 
+lazy val bridge = (project in file("bridge"))
+  .dependsOn(firrtl)
+  .settings(commonSettings)
+  .settings(
+    libraryDependencies ++= Seq("org.chipsalliance" %% "chisel" % "6.0.0",
+      "org.apache.commons" % "commons-lang3" % "3.12.0",
+      "org.apache.commons" % "commons-text" % "1.9"),
+    addCompilerPlugin("org.chipsalliance" % "chisel-plugin" % "6.0.0" cross CrossVersion.full),
```
Comment (ekiwi) [inline build.sbt:159]: I don't really want this project to depend on `chisel6` which is why am hesitant to upstream this code.

Comment (jerryz123) [reply build.sbt:159]: Is it ok if only the `bridges` project depends on `chisel6`?

Comment (ekiwi) [reply build.sbt:159]: oh sorry, I wasn't paying attention.

Yes. It is ok to have a separate bridges project. I won't publish it to maven though. But I can help you publish it if you want to.

Comment (jerryz123) [reply build.sbt:159]: Thanks. I need to do a bit more work with this before we merge. 

Maybe publish in the future, but it isn't necessary yet.

Comment (ekiwi) [reply build.sbt:159]: OK. Please ping we when you want to merge.

**Review by ekiwi — APPROVED** [review]

**Pull request merged**